lib/utils.ts

```diff
@@ -395,6 +419,43 @@ async function requestAsync(
 		opts.headers = nativeHeaders;
 	}
 
+	const form = getForm();
+	let hasBlob = false;
+	for (const [k, v] of Object.entries<string | WebResourceFile>(
+		options.body ?? {},
+	)) {
+		if (isFile(v)) {
+			hasBlob = true;
+			form.append(k, v, v.name);
+		} else {
+			form.append(k, v);
+		}
+	}
+
+	if (hasBlob) {
+		if (IS_BROWSER) {
```

Comment:
If you do
```suggestion
		if (form instanceof FormData) { // or the other way around `instanceof FormDataNodeType` if you prefer
```
then you should be able to remove the `form as FormData` & the `form as FormDataNodeType` casts (and reduce the `IS_BROWSER` checks).https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax

Reply:
I am not sure this works @thgreasi as FormDataNode is also a FormData